File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int n) {
        
    List<List<Integer>> pascal =new ArrayList();
        for(int i=0;i<n;i++)
        {
            List<Integer> row =new ArrayList();
            
            for(int j=0;j<=i;j++)
            {
                if(j==0|| j==i)
                {
                    row.add(1);
                }
                else
                {
                    List<Integer> prev= pascal.get(i-1);
                    row.add(prev.get(j)+prev.get(j-1));
                    
                }
            }
            pascal.add(row);
            
        }
        return pascal;
        
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<int> solve(int row)
    {
        vector<int> temp;
        temp.push_back(1);
        long long res=1;
        for(int col=1;col<row;col++)
        {
            res=res*(row-col)/col;
            temp.push_back(res);
        }
        return temp;
    }
    
    
    vector<vector<int>> generate(int n) {
        vector<vector<int>> pascal;
        for(int i=1;i<=n;i++)
        {
            pascal.push_back(solve(i));
        }
        
        
        
        
        
        return pascal;
    }
};